### PR TITLE
Draft for unified build script

### PR DESCRIPTION
**Pull request by SinTh0r4s**

### CODE REVIEW

**Comment by SinTh0r4s** [comment]
Tested with HydroEnergy in dev and it works just fine. Before it can be merged it MUST be testet with all NEI addons!

**Comment by SinTh0r4s** [comment]
It required one more patch to allow shadow in dev and sources jar

File: build.gradle [hunk 1]
```diff
@@ -1,40 +1,57 @@
...
+import java.util.concurrent.TimeUnit
+
 buildscript {
     repositories {
-        mavenCentral()
-        jcenter()
         maven {
-            name = "gt"
-            url = "https://gregtech.overminddl1.com/"
+            name = "jitpack"
```
Comment (mitchej123) [inline build.gradle:17]: jitpack should be lower in the list?

File: build.gradle [hunk 2]
```diff
@@ -44,121 +61,490 @@ idea {
...
+    maven {
+        name = "Overmind forge repo mirror"
+        url = "https://gregtech.overminddl1.com/"
+    }
+    if(usesMixins.toBoolean()) {
+        maven {
+            name = "sponge"
+            url = "https://repo.spongepowered.org/repository/maven-public"
+        }
+        maven {
```
Comment (mitchej123) [inline build.gradle:200]: only adding jitpack if it uses mixins?

**Review by mitchej123 — APPROVED** [review]
Can you confirm this works in the full GTNH pack?

A few questions on the gradle file otherwise as long as it's tested looks fine.

**Comment by SinTh0r4s** [comment]
> Can you confirm this works in the full GTNH pack?

Yes I can

File: build.gradle (hunk 1 above)
Comment (SinTh0r4s) [reply build.gradle:17]: The build script is loaded from the example mod repo. Please open an issue there

File: build.gradle (hunk 2 above)
Comment (SinTh0r4s) [reply build.gradle:200]: The build script is loaded from the example mod repo. Please open an issue there

File: build.gradle (hunk 1 above)
Comment (mitchej123) [reply build.gradle:17]: I've mentioned this before.  And regardless of where it's from, I'm reviewing for the NEI repo, jitpack being that high isn't appropriate for building NEI.

**Comment by SinTh0r4s** [comment]
Ready to merge

**Comment by mitchej123** [comment]
Feel free to merge after rebasing on master.  Thanks for the work!

**Pull request merged**